Update regions_by_endpoint for AWS sdk upgrade.

ywangd opened the pull request:
Also add test to ensure the file has at least one entry for each region so that it is easy to spot missing regions in future upgrades.

Relates: #131050
Resolves: #131392

ywangd committed:
Update regions_by_endpoint for AWS sdk upgrade.

Also add test to ensure the file has at least one entry for each region
so that it is easy to spot missing regions in future upgrades.

Relates: #131050

elasticsearchmachine commented:
Pinging @elastic/es-distributed-coordination (Team:Distributed Coordination)

ywangd committed:
Merge remote-tracking branch 'origin/main' into es-131392-fix

nicktindall reviewed (approved):
Nice safeguard going forward!

DaveCTurner reviewed (changes requested):
Hang on (I'll explain why in a sec)

DaveCTurner reviewed (approved):
Actually hmm this is tricky.

The `RegionFromEndpointGuesser` is there for bwc with the pre-8.19 behaviour in which you could omit the region, but my thinking was that brand-new regions don't need this bwc. Moreover if we try and keep the list up to date we'll have to keep on upgrading the AWS SDK rather more often than we would otherwise.

However on reflection I think we have to do this anyway because I suspect the v1 SDK will pick up these new regions automatically, so a user on 8.18 in the future may have working region-free config that won't work on upgrade to the v2 SDK unless we keep the v2 SDK up to date.

So, ok to proceed, but some concerns here about the implications.

ywangd commented:
Thanks for sharing the thoughts, David. Very helpful. I guess we probably can still call this best-effort to provide v1 users better upgrading experience, i.e. we strive to not break existing "region free" configs, but it may not be done continiously for every minor sdk version and the fallback (and actually the right solution) for users is to specify the region.

ywangd commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'main' into es-131392-fix

elasticsearchmachine committed:
Mute org.elasticsearch.packaging.test.DockerTests test072RunEsAsDifferentUserAndGroup #131412

kanoshiou committed:
ESQL: Fix inconsistent column order in MV_EXPAND (#129745)

The new attribute generated by MV_EXPAND should remain in the original position. The projection added by ProjectAwayColumns does not respect the original order of attributes.

Make ProjectAwayColumns respect the order of attributes to fix this.

jan-elastic committed:
ES|QL categorize options (#131104)

* ES|QL categorize options

* refactor options

* fix serialization

* polish

* add verfications

* better test coverage + polish code

* better test coverage + polish code

ldematte committed:
Migrate x-pack-autoscaling REST tests (#131365)

This PR migrates legacy rest tests in the x-pack autoscaling module

pquentin committed:
Remove 'index' from snapshot clear_cache query params (#131067)

It's already part of the path parts, it's not useful to duplicate it in query
parameters.

iverase committed:
[DiskBBQ] Use PackedLongValues to hold offsets on heap while writing (#131411)

ywangd committed:
Merge remote-tracking branch 'origin/main' into es-131392-fix

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [9.1](https://github.com/elastic/elasticsearch/pull/131450)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/131450">](https://github.com/elastic/elasticsearch/pull/131450) |
| ✅ |  [8.19](https://github.com/elastic/elasticsearch/pull/131451)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/131451">](https://github.com/elastic/elasticsearch/pull/131451) |
